### ISSUE TITLE
Controller

### ISSUE THREAD

**Issue by willuk2010** [ChanceM/pfSense-pkg-zerotier#11]
Hey @ChanceM 

Good work on this, i managed to build and deploy this on my pfSense router this evening, and i am able to connect to networks fine on the Zerotier dashboard.

But i'm trying to get the Controller working, i'm able to set up a controller network, but the client (on my iphone) just shows Network not Found.

I must be missing something, or my understanding of how the controller works.. my assumption was i could create a controller network from pfSense, connect my clients (phone etc), and allow access to my devices behind the pfSense firewall.

Cheers!

**Comment by ChanceM** [comment]
I will have to spin up my VM and do some tests. I apologize for the delay in response just coming back from a vacation. You should get a zt interface that you can assign and then you will need to manually set the IP to the one given from zt and add rules allowing traffic.

**Comment by knightian** [comment]
@ChanceM hello, thanks for this. How would we go about updating the package to v1.8.1?

**Comment by ChanceM** [comment]
> @ChanceM hello, thanks for this. How would we go about updating the package to v1.8.1?

I believe 1.6.6 is the latest available FreeBSD 12.

**Comment by knightian** [comment]
> So testing this, creating a network does in fact create a tap interface, but I only see one. Joining the network from the same device creates a ZT interface.

I have a zt interface and a tap interface after joining the network from the pfsense device. I assume we only use the zt interface set it up with the IP assigned from the controller and such? I'm getting direct connection to the controller but all other nodes have to relay to it and I can't ping in.

**Comment by knightian** [comment]
> > @ChanceM hello, thanks for this. How would we go about updating the package to v1.8.1?
> 
> I believe 1.6.6 is the latest available FreeBSD 12.

Yea seems that way, so after some research i learned how to compile 1.8.8 and built my own pkg for it.

Link to my pkg if anyone else wants it https://we.tl/t-tNOBIrdV9B

**Comment by knightian** [comment]
> > So testing this, creating a network does in fact create a tap interface, but I only see one. Joining the network from the same device creates a ZT interface.
> 
> I have a zt interface and a tap interface after joining the network from the pfsense device. I assume we only use the zt interface set it up with the IP assigned from the controller and such? I'm getting direct connection to the controller but all other nodes have to relay to it and I can't ping in.

Can confirm that at least when not running a controller, it's safe to ignore the tap interface that shows up, just mount the ztxxxxxxxxxx interface that shows up and give it an IP in your zt network, assign it that same IP at the controller and then make your firewall rules correct and you're good :D

**Comment by fxn2020** [comment]
@knightian Your shared file has expired, the latest is 1.83, can you recompile and share it?

**Comment by knightian** [comment]
> @knightian Your shared file has expired, the latest is 1.83, can you recompile and share it?

Heya the official package has been updated to 1.8.3 so if you use the url in the readme and change 1.6.6 to 1.8.3 in the url, it will add the latest official package.

**Comment by fxn2020** [comment]
@knightian I really don't have the energy to build the environment needed for compilation. Could you share the pfsense-pkg-zerotier.txz you compiled, thank you in advance. My email: fxneng@gmail.com

**Comment by brahmanggi** [comment]
> > > @ChanceM hello, thanks for this. How would we go about updating the package to v1.8.1?
> > 
> > 
> > I believe 1.6.6 is the latest available FreeBSD 12.
> 
> Yea seems that way, so after some research I learned how to compile 1.8.1 and build my own pkg for it. Working on my pfsense (I looked at the file structure of the v1.6.6 pkg and copied that in my pkg from the compiled bins)
> 
> Link to my pkg if anyone else wants it https://we.tl/t-tNOBIrdV9B
> 
> <img alt="Screen Shot 2021-11-13 at 4 50 15 pm" width="1166" src="https://user-images.githubusercontent.com/72685601/141607580-5e1785e3-fe85-47c5-8458-95413fbe1adc.png">

hello @knightian the link expired I want to try it on my machine, thank you in advance

**Comment by fxn2020** [comment]
I compiled one by myself, you can download it if you need it。[pfsense-pkg-zerotier](https://pfschina.org/wp/wp-content/uploads/2021/10/pfSense-pkg-zerotier.txz_.zip)，[zerotier1.83.txz](https://pfschina.org/wp/wp-content/uploads/2019/09/zerotier-1.8.3.txz_.zip)

**Comment by brahmanggi** [comment]
> I compiled one by myself, you can download it if you need it。[pfsense-pkg-zerotier](https://pfschina.org/wp/wp-content/uploads/2021/10/pfSense-pkg-zerotier.txz_.zip)，[zerotier1.83.txz](https://pfschina.org/wp/wp-content/uploads/2019/09/zerotier-1.8.3.txz_.zip)

thank you very much, do I need to compiling again or is it already pkg files?

**Comment by fxn2020** [comment]
@brahmanggi It has been compiled, downloaded and decompressed, uploaded to pfsense, installed in the shell environment, command: pkg install pfsense-pkg-zerotier.txz

**Comment by brahmanggi** [comment]
> @brahmanggi It has been compiled, downloaded and decompressed, uploaded to pfsense, installed in the shell environment, command: pkg install pfsense-pkg-zerotier.txz

yes, I will try to upload to my pfbox, and again thank you very much

**Comment by knightian** [comment]
Just be mindful if you remove the package, it keeps the config for the zt interface still in the config.xml and next boot the device gets blocked asking you to reassign the interfaces because it can no longer find the zt interface that it has a config specified for.

This is something I encountered during upgrade. So if you are remote accessing the box or have no local video/serial console access then best you never uninstall the package once installed.

**Comment by ChanceM** [comment]
@knightian I'm looking at this. I think I can remove the interfaces on deinstall to hopefully prevent this.

**Comment by knightian** [comment]
> @knightian I'm looking at this. I think I can remove the interfaces on deinstall to hopefully prevent this.
> 
> Updated Package for Install: https://app.box.com/s/m3a6m081d1gjpxrwob6rktfilrktsrnn

Thanks, does the new package you posted have this fix?

**Comment by ChanceM** [comment]
@knightian not yet this addition shows the interface associated with the network or a link to assign an interface for a new network, but that was just the first step getting the associated interface so I can remove them on deinstall.

**Comment by OvrAp3x** [comment]
This is great work!
Appriciate it :-)

**Comment by knightian** [comment]
@ChanceM is that issue of the interface getting removed and screwing up pfSense updates being worked on? it makes using the zerotier pkg in the field untenable because whenever we update pfSense we would have to visit the site and physically reset up the pfSense device :(

**Comment by ChanceM** [comment]
@knightian I would love to say yes, but the reality is I do not have a whole lot of time to devote to it. I did dig a little bit into it tonight, but nothing new to report.

**Comment by Opnwall** [comment]
@ChanceM In pfSense plus 23.01, the plug-in can no longer be installed and used normally, and I hope it can be updated in time
